src/input/input_python.rs

```diff
@@ -302,35 +302,40 @@ impl<'a> Input<'a> for PyAny {
         }
     }
 
-    fn ultra_strict_float(&self) -> ValResult<f64> {
+    fn ultra_strict_float(&'a self) -> ValResult<EitherFloat<'a>> {
```

Comment:
I think with this change we might be able to remove `ultra_strict_float` completely, maybe that's too much work for this PR.